### PR TITLE
Add ops(max_pool3d and max_pool3d_with_indices) | feat(atenlib)

### PR DESCRIPTION
**Pull request by fatcat-z**
1. Add max_pool3d and max_pool3d_with_indices ops.
2. Refactor the code so max_pool2d and max_pool3d could share the same helper function. And move them to nn.py file.
3. Refactor the code so max_pool2d_with_indices and max_pool3d_indices could share the same helper function. And move them to nn.py file.
4. Add tests.

### CODE REVIEW

**Review by xiaowuhu — APPROVED** [review]

File: onnxscript/function_libs/torch_aten/ops/nn.py [hunk 1]
```diff
@@ -872,17 +863,135 @@ def aten_max_pool2d_with_indices_backward(
...
+    # easy but hacky way to get flattened indices values
+    # to be used to convert the indices values to non-flattened.
+    # In ONNX the indices are computed as a flatten 1-D tensor,
+    # so the values in indices are in [0, N x C x D1 x ... x Dn).
+    # To convert the indices to the same format used by Pytorch,
+    # we first execute a maxpool with a kernel and stride of 1 on the same input.
+    # This will result in a tensor of indices in which each index will have it's own value.
+    # Using this tensor as a reference, we extract the first index of each axis and subtract
+    # it from each index of this axis in the indices to convert.
+    # This step will result in a tensor were each dimension has values of indices within
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/nn.py:956]:
```suggestion
    # This step will result in a tensor where each dimension has values of indices within
```
nit

File: onnxscript/function_libs/torch_aten/ops/nn.py [hunk 2]
```diff
@@ -872,17 +863,135 @@ def aten_max_pool2d_with_indices_backward(
...
+    # the dimension it is in.
+    # For Maxpool1d(kernel=1,stride=1,return_indices=True), with the input torch.ones(1,2,2).
+    # The computed indices are the following:
+    # output indices pytorch :
+    #     [[0,1],
+    #     [0,1]]
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/nn.py:962]:
```suggestion
    #      [0,1]]
```

File: onnxscript/function_libs/torch_aten/ops/nn.py [hunk 3]
```diff
@@ -872,17 +863,135 @@ def aten_max_pool2d_with_indices_backward(
...
+    pool_result, indices = op.MaxPool(
+        self,
+        ceil_mode=ceil_mode,
+        dilations=dilation,
+        kernel_shape=kernel_size,
+        pads=padding,
+        strides=stride,
+    )
+
+    # easy but hacky way to get flattened indices values
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/nn.py:947]:
```suggestion
    # Simple but hacky way to get flattened indices values
```

File: onnxscript/function_libs/torch_aten/ops/nn.py [hunk 4]
```diff
@@ -872,17 +863,135 @@ def aten_max_pool2d_with_indices_backward(
...
+    #     [[0,1],
+    #     [2,3]]
+    # Then call slick to get the first value of each line (so 0 and 2).
+    # And the subtraction executes :
+    #     [[0-0,1-0],
+    #     [2-2,3-2]]
+    # So indices results to the expected output which is :
+    #     [[0,1],
+    #     [0,1]]
+    # For more information :
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/nn.py:977]: Thanks for the explanation! This makes a lot of sense

File: onnxscript/function_libs/torch_aten/ops/nn.py [hunk 5]
```diff
@@ -872,17 +863,135 @@ def aten_max_pool2d_with_indices_backward(
...
+    # https://github.com/pytorch/pytorch/pull/16455#issuecomment-460776407
+    _, flatten_indices = op.MaxPool(
+        self, dilations=dilation, kernel_shape=n_dims_one, strides=n_dims_one
+    )
+
+    ends_t = op.Constant(value_ints=n_dims_one)
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/nn.py:983]:
```suggestion
    ends = op.Constant(value_ints=n_dims_one)
```

maybe just

File: onnxscript/tests/function_libs/torch_aten/extra_opinfo.py [hunk 6]
```diff
@@ -163,9 +163,9 @@ def sample_inputs_convolution(op_info, device, dtype, requires_grad, **kwargs):
         )
 
 
-def sample_inputs_layer_norm(
-    op_info, device, dtype, requires_grad, **kwargs  # pylint: disable=unused-argument
-):
+def sample_inputs_layer_norm(op_info, device, dtype, requires_grad, **kwargs):
+    del op_info
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_aten/extra_opinfo.py:167]:
```suggestion
    del op_info  # unused
```

**Review by justinchuby — APPROVED** [review]

**Pull request merged**